### ISSUE TITLE
Problem with generating code when a type uses the query type as field when Modelsuffix is set

### ISSUE THREAD

**Issue by xtroce** [kobylynskyi/graphql-java-codegen#176]
## Issue Description

While using maven to generate the GraphQL data generated form an inspection query, there is a problem when using the Query object in one of the internal types and using Model suffix.

However the remote API has one type: Job which has as an internal variable a type Query. Since query is a special type the generated class does not get the suffix added. However in the class generated that uses the query as field gets the suffix added to the class name, which leads to uncompileable  code since the object cannot be found.

```graphql
schema {
  query: Query
  mutation: Mutation
}

"""
A job corresponds to some long running task that the client should poll for status.
"""
type Job {
  """This indicates if the job is still queued or has been run."""
  done: Boolean!

  """Globally unique identifier."""
  id: ID!

  """
  This field will only resolve once the job is done. Can be used to ask for object(s) that have been changed by the job.
  """
  query: Query
}
```
(i did not paste query itself since it's huge see this pastebin: https://pastebin.com/SfcLRLdf for the whole schema file.  )

```java
public class JobMysuffix {

    @javax.validation.constraints.NotNull
    private Boolean done;
    @javax.validation.constraints.NotNull
    private String id;
    private QueryMySuffix query;
    ....
}
```
## Expected Result

Either also add the suffix to the query, or remove the suffix from objects that are of type Query

## Actual Result
in the example below class Job uses     private QueryMySuffix query; but the query class is just called Query.

## Your Environment and Setup

* **graphql-java-codegen**: *1.8.1*
* **Build tool**: *Maven*
* **Java tool**: *openJDK 14*
* **Mapping Config**: *default config with modelNameSuffix set* 

```xml
                        <configuration>
                            <graphqlSchemaPaths>
                                <graphqlSchemaPath>${project.basedir}/src/main/resources/schema.graphqls</graphqlSchemaPath>
                            </graphqlSchemaPaths>
                            <outputDir>${project.basedir}/src/gen/java</outputDir>
                            <packageName>test.graphql</packageName>
                            <apiPackageName>test.graphql.api.client</apiPackageName>
                            <modelPackageName>test.graphql.model.client</modelPackageName>
                            <customTypesMapping>
                                <Date>java.util.Date</Date>
                                <DateTime>java.util.Date</DateTime>
                                <Decimal>java.math.BigDecimal</Decimal>
                                <FormattedString>java.lang.String</FormattedString>
                                <HTML>java.lang.String</HTML>
                                <JSON>org.json.JSONObject</JSON>
                                <Money>java.math.BigDecimal</Money>
                                <StorefrontID>java.lang.String</StorefrontID>
                                <UnsignedInt64>java.lang.Long</UnsignedInt64>
                                <URL>java.net.URI</URL>
                                <UtcOffset>java.lang.String</UtcOffset>
                            </customTypesMapping>
                            <modelNameSuffix>MySuffix</modelNameSuffix>
                            <requestSuffix>MyRequestSuffix</requestSuffix>
                            <generateRequests>true</generateRequests>
                        </configuration>
```

**Comment by kobylynskyi** [comment]
Hey @xtroce 
To be honest, I haven't seen a schema that has a reference to `type Query` from another type.
So if you really need this, I will generate another model object type for each of the reserved types: `Query`, `Mutation`, `Subscription` (considering prefix/suffix and other options for model-generation)

**Comment by xtroce** [comment]
The problem stems from the fact that the graphqls is generated by an inspection query. It would also not be my choice to built up an API returning Query objects, but it is something I came across in the wilderness and afaik it is not forbidden by the specs, so I thought I fill a bug report here.  
Especially since this tool is going to be used to generate clients, I could imagine people running into this problem in the future, with graphQL getting more traction. Since it is not my own API I would of course be glad if the problem would be fixed. ;)
I could try to spend some time in my free time to implement the change, since hey, it's open source and if I can help out , why not. If you however say you exactly know where to change it and it does not cost you tons of work I would of course be glad.

I worked around it for now by just changing the type to a string in the schema.graphqls, since I do not need to use this part of the API atm, and don't want to change the variables every time I regenerate the code, but this will however break the possibility for me to download the file from the inspection query and directly process it + using this plugin if I ever want to query this part of the API without disabling/enabling code generation manually every time.

FYI I am trying to use the shopify GraphQL API. They have the possibility to run internal jobs, which in turn you can query to get a query back which shows you what is changed.

see: https://shopify.dev/docs/admin-api/graphql/reference/object/job?api[version]=2020-04

**Comment by kobylynskyi** [comment]
@xtroce thanks for your input.
One extra note: if we generate POJO classes for Query/Mutation/Subscription types then we will need to generate `Resolver` classes for each operation (because by default `generateParameterizedFieldsResolvers` equal to `true`).
For example, for [your schema](https://pastebin.com/SfcLRLdf) we will need to generate 150 extra classes: 2 POJOs (`Query.java` and `Mutation.java`) and 148 `Resolver` classes for parametrized fields (62 `XXXResolver.java`and 86 `XXXMutation.java`).

**Comment by kobylynskyi** [comment]
@xtroce I've created #206 to work on new config options to support client-side codegen for the schema you provided.
Please take a look and let me know what you think. Thanks!

**Comment by kobylynskyi** [comment]
The solution for this bug will be released as part of 2.1.0 this week.
